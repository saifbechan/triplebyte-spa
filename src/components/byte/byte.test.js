import React from 'react';
import { Provider } from 'react-redux';
import { render, fireEvent } from '@testing-library/react';
import configureStore from 'redux-mock-store';

import Byte from './byte.component';

const mockStore = configureStore([]);

let store;
let renderedByte;

const byteValue = 1;
const idx = 5;
const toggleByte = jest.fn();

beforeEach(() => {
  store = mockStore({
    myState: 'mystate'
  });

  store.dispatch = jest.fn();

  renderedByte = render(
    <Provider store={store}>
      <Byte idx={idx} toggleByte={toggleByte}>
        {byteValue}
      </Byte>
    </Provider>
  );
});

it('renders children correctly', () => {
  const { asFragment } = renderedByte;
  expect(asFragment()).toMatchSnapshot();
});

it('calls dispatch correctly on click', () => {
  const { getByText } = renderedByte;
  fireEvent.click(getByText(new RegExp(byteValue, 'i')));
  expect(store.dispatch).toHaveBeenCalledWith({
    payload: idx,
    type: 'TOGGLE_BYTE'
  });
});
