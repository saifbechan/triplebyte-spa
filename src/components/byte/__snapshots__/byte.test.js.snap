// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders children correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-AykKC hZQdx"
  >
    1
  </div>
</DocumentFragment>
`;
